File: Algo/Leetcode/477HammingDistanceSum.cpp
// count the 1's num at each pos. Difference at each pos is c * (n-c)
class Solution {
public:
    int totalHammingDistance(vector<int>& nums) {
        int count = 0;
        for(int i=0; i<32; i++){
            int c = 0;
            for(auto n:nums) c += (n>>i) & 1;
            count += c * (nums.size() - c);
        }
        return count;
    }
};

File: Algo/Leetcode/477HammingDistanceSum.py
class Solution(object):
    def totalHammingDistance(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        n = len(nums)
        diff = 0
        for i in range(32):
            c = 0
            for num in nums:
                c += 1 if (num & 1 << i) > 0 else 0
            diff += (n - c) * c

        return diff
